src/Runner.Common/ResultsServer.cs

```diff
@@ -94,5 +117,138 @@ public void InitializeResultsClient(Uri uri, string token)
+                    Trace.Error(ex);
+                    retries++;
+                    this._websocketClient = null;
+                }
+            } while (retryConnection && !connected && retries < 3);
+        }
+
+        public async Task<bool> AppendLiveConsoleFeedAsync(Guid scopeIdentifier, string hubName, Guid planId, Guid timelineId, Guid timelineRecordId, Guid stepId, IList<string> lines, long? startLine, CancellationToken cancellationToken)
+        {
+            if (_websocketConnectTask != null)
+            {
+                await _websocketConnectTask;
+            }
+
+            bool delivered = false;
+            int retries = 0;
+
+            // "_websocketClient != null" implies either: We have a successful connection OR we have to attempt sending again and then reconnect
+            // ...in other words, if websocket client is null, we will skip sending to websocket
+            if (_websocketClient != null)
+            {
+                var linesWrapper = startLine.HasValue
+                    ? new TimelineRecordFeedLinesWrapper(stepId, lines, startLine.Value)
+                    : new TimelineRecordFeedLinesWrapper(stepId, lines);
+                var jsonData = StringUtil.ConvertToJson(linesWrapper);
+                var jsonDataBytes = Encoding.UTF8.GetBytes(jsonData);
+                // break the message into chunks of 1024 bytes
+                for (var i = 0; i < jsonDataBytes.Length; i += 1 * 1024)
+                {
+                    var lastChunk = i + (1 * 1024) >= jsonDataBytes.Length;
+                    var chunk = new ArraySegment<byte>(jsonDataBytes, i, Math.Min(1 * 1024, jsonDataBytes.Length - i));
+
+                    while (!delivered && retries < 3)
+                    {
+                        try
+                        {
+                            await _websocketClient.SendAsync(chunk, WebSocketMessageType.Text, endOfMessage: lastChunk, cancellationToken);
+                            delivered = true;
+                        }
+                        catch (Exception ex)
+                        {
+                            if (_websocketClient != null)
+                            {
+                                var delay = BackoffTimerHelper.GetRandomBackoff(MinDelayForWebsocketReconnect, MaxDelayForWebsocketReconnect);
+                                Trace.Info($"Websocket is not open, let's attempt to connect back again with random backoff {delay} ms.");
+                                Trace.Error(ex);
+                                retries++;
+                                InitializeWebsocketClient(_liveConsoleFeedUrl, _token, delay);
+                            }
+                        }
+                    }
+                }
+            }
+
+            if (!delivered)
+            {
+                // Giving up for good if we still can't deliver a message after 3 tries
+                _websocketClient = null;
```

Comment:
should we give up or keep retrying?
assume there is a network blip at the beginning of the job, then you won't get any live console log throughout the entire job even after the network blip recovery.

Reply:
I added a timer to track the last failure - so if 10 minutes passed, we will try again.